taxcalc/tests/test_calculate.py

```diff
@@ -285,12 +286,47 @@ def test_diagnostic_table():
     tax_dta.flpdyr += 17
     puf = Records(tax_dta, weights = weights)
     # Create a Calculator
-    
+
     calc = Calculator(params=params, records=puf, sync_years=False)
 
     calc.diagnostic_table()
 
 
+def test_aggregate_corporate_income_tax_vals():
+    # we need the records' year at 2008 for blow up step. So param's year needs to be 2008 to past the test
+    irates = {2008:0.015, 2009:0.020, 2010:0.022, 2011:0.020, 2012:0.021,
+          2013:0.022, 2014:0.023, 2015:0.024, 2016:0.024, 2017:0.024,
+          2018:0.024, 2019:0.024}
+    params = Parameters(start_year=2008, inflation_rates=irates)
+    # Create a Public Use File object
+    puf = Records(tax_dta)
+    # Create a Calculator
+    calc = Calculator(params=params, records=puf)
+
+    agg_e_and_p = ((calc.records.e02000 * calc.records.s006).sum()
+                   + (calc.records.e09400 * calc.records.s006).sum())
+
+    agg_dividends = (calc.records.e00600 * calc.records.s006).sum()
+
+    agg_capgains = ((calc.records.e23250 * calc.records.s006).sum() +
+                    (calc.records.e22250 * calc.records.s006).sum() +
+                    (calc.records.e23660 * calc.records.s006).sum())
+
+    agg_bonds = ((calc.records.e00400 * calc.records.s006).sum()
+                 + (calc.records.e00300 * calc.records.s006).sum())
+
+    agg_compensation = ((calc.records.e00200 * calc.records.s006).sum() +
+                        (calc.records.e00250 * calc.records.s006).sum() +
+                        (calc.records.e33420 * calc.records.s006).sum() +
+                        (calc.records.e03600 * calc.records.s006).sum() +
+                        (calc.records.e07240 * calc.records.s006).sum())
+
+    assert abs(calc.records.agg_comp - agg_compensation) < .001
```

Comment:
I am curious - where are these aggregates used? I don't recall them being 
reported on the web interface. Also, I am concerned that if do a large 
number of these (especially treating the sumation as a action on an 
object) it will slow things down.

dan

On Wed, 1 Jul 2015, mmessick wrote:

> 
> In taxcalc/tests/test_calculate.py:
> 
> > +    agg_dividends = (calc.records.e00600 * calc.records.s006).sum()
> > +
> > +    agg_capgains = ((calc.records.e23250 * calc.records.s006).sum() +
> > +                    (calc.records.e22250 * calc.records.s006).sum() +
> > +                    (calc.records.e23660 * calc.records.s006).sum())
> > +
> > +    agg_bonds = ((calc.records.e00400 * calc.records.s006).sum()
> > +                 + (calc.records.e00300 * calc.records.s006).sum())
> > +
> > +    agg_compensation = ((calc.records.e00200 * calc.records.s006).sum() +
> > +                        (calc.records.e00250 * calc.records.s006).sum() +
> > +                        (calc.records.e33420 * calc.records.s006).sum() +
> > +                        (calc.records.e03600 * calc.records.s006).sum() +
> > +                        (calc.records.e07240 * calc.records.s006).sum())
> > +
> > +    assert abs(calc.records.agg_comp - agg_compensation) < .001
> 
> I've tried tightening it up, and haven't gotten good results. Not really sure why
> 
> —
> Reply to this email directly or view it on
> GitHub.[AHvQVbIo7EN2cy-vTvCzib1ZbTKNLXn6ks5oZDJNgaJpZM4FES6E.gif]
> 
> 
>